### ISSUE TITLE
Running with Nginx?

### ISSUE THREAD

**Issue by callowaysutton** [pmmp/PocketMine-MP#2680]
Wouldn't it be feasible to run this in any webserver since it's just PHP? And why not use run it in a web server over running it on the internal PHP server which is never intended to be used in production for it's security bugs and instabilities. It'd make the most sense to run it on something like Nginx's PHP-FPM approach because of the amount of control you have over the server (load balancing, task scheduling, and the like).

Just a thought I had, should I make a pull request for this?

**Comment by SOF3** [comment]
PocketMine uses PHP CLI. PHP-FPM is supposed to die in up to a few minutes, and PHP CLI is supposed to run longer.
Extensions like pthreads would not be available in web servers.
The load balancing in Nginx is about balancing multiple HTTP requests. Clients connect to PocketMine using the Minecraft UDP protocol, not through HTTP. They are totally irrelevant.
The load balancing in Nginx works by starting a new PHP runtime for every client request. This is apparently entirely different from PocketMine's system.
In short, if you want to experiment running a Minecraft server in nginx, just rewrite another project; it could almost reuse no any infrastructure from PocketMine, which heavily relies on using the same PHP runtime. (Although I am pretty sure you won't be successful doing that)

**Comment by renay** [comment]
> PocketMine uses PHP CLI. PHP-FPM is supposed to die in up to a few minutes, and PHP CLI is supposed to run longer.
> Extensions like pthreads would not be available in web servers.
> The load balancing in Nginx is about balancing multiple HTTP requests. Clients connect to PocketMine using the Minecraft UDP protocol, not through HTTP. They are totally irrelevant.
> The load balancing in Nginx works by starting a new PHP runtime for every client request. This is apparently entirely different from PocketMine's system.
> In short, if you want to experiment running a Minecraft server in nginx, just rewrite another project; it could almost reuse no any infrastructure from PocketMine, which heavily relies on using the same PHP runtime. (Although I am pretty sure you won't be successful doing that)

Pthreads is no longer relevant, will PocketMine be rewritten to parallel?
https://github.com/krakjoe/parallel
